pkg/api/openshiftversion.go

```diff
@@ -0,0 +1,17 @@
+package api
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// OpenShiftVersion represents an OpenShift version that can be installed
+type OpenShiftVersion struct {
+	MissingFields
+
+	ID       string `json:"id,omitempty"`
+	Deleting bool   `json:"deleting,omitempty"` // https://docs.microsoft.com/en-us/azure/cosmos-db/change-feed-design-patterns#deletes
```

Comment:
@hawkowl for what use we need to track which version is being deleted?

Reply:
> The change feed does not capture deletes. If you delete an item from your container, it is also removed from the change feed. The most common method of handling this is adding a soft marker on the items that are being deleted. You can add a property called "deleted" and set it to "true" at the time of deletion. This document update will show up in the change feed. You can set a TTL on this item so that it can be automatically deleted later.

Reply:
Yep. This isn't used right now, but I can foresee us perhaps using the changefeed to update/prune/etc images or something, so, might as well keep it.